js/src/dropdown.js

```diff
@@ -176,7 +178,14 @@ class Dropdown extends BaseComponent {
         referenceElement = this._config.reference
       }
 
-      this._popper = Popper.createPopper(referenceElement, this._menu, this._getPopperConfig())
+      const popperConfig = this._getPopperConfig()
+      const isDisplayStatic = popperConfig.modifiers.find(modifier => modifier.name === 'applyStyles' && modifier.enabled === false)
+
+      this._popper = Popper.createPopper(referenceElement, this._menu, popperConfig)
+
+      if (isDisplayStatic) {
+        Manipulator.setDataAttribute(this._menu, 'popper', 'static')
+      }
```

Comment:
Setting the `data-bs-popper` attribute after creating the popper's instance makes sure that the popper's instance has been created and then styles are applied. So that popper does not warn about the styles and places the elements well.